conceptnet5/builders/reduce_assoc.py

```diff
@@ -136,35 +131,66 @@ def make_filtered_concepts(filename, cutoff=3, en_cutoff=3):
 
 
 def make_conceptnet_association_graph(
```

Comment:
Is there a reason why this shouldn't happen in the constructor for `ConceptNetAssociationGraph`?

Reply:
This factory function calls the add_edge method, which is extended in the subclasses of the graph class to store additional edge data in data members not present in the base class.  If this code were moved into the constructor of the base class, those members would not be present during base class construction, so construction would fail.  I don't see an obvious way around this, and duplicating the code in two subclass constructors seems undesirable to me.  We could make this factory function a classmethod of the base class, if that strikes you as better.

Reply:
I think I'd like that more since it makes this code more maintainable.